cloudcix_primitives/storage_hyperv.py

```diff
@@ -76,50 +87,65 @@ def run_host(host, prefix, successful_payloads):
-            if ret["channel_code"] != CHANNEL_SUCCESS:
-                return False, fmt.channel_error(ret, f"{prefix+2}: " + messages[prefix + 2]), fmt.successful_payloads
-            if ret["payload_code"] != SUCCESS_CODE:
-                return False, fmt.payload_error(ret, f"{prefix+3}: " + messages[prefix + 3]), fmt.successful_payloads
-            fmt.add_successful('create_storage_file', ret)
+        ret = rcc.run(payloads['create_storage_file'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+2}: " + messages[prefix + 2]), fmt.successful_payloads
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f"{prefix+3}: " + messages[prefix + 3]), fmt.successful_payloads
+        fmt.add_successful('create_storage_file', ret)
+
+        ret = rcc.run(payloads['prepare_storage_file'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+4}: " + messages[prefix + 4]), fmt.successful_payloads
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f"{prefix+5}: " + messages[prefix + 5]), fmt.successful_payloads
+        fmt.add_successful('prepare_storage_file', ret)
+
+        ret = rcc.run(payloads['dismount_storage_file'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+6}: " + messages[prefix + 6]), fmt.successful_payloads
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f"{prefix+7}: " + messages[prefix + 7]), fmt.successful_payloads
+        fmt.add_successful('dismount_storage_file', ret)
 
-            ret = rcc.run(payloads['dismount_storage_file'])
-            if ret["channel_code"] != CHANNEL_SUCCESS:
-                return False, fmt.channel_error(ret, f"{prefix+4}: " + messages[prefix + 4]), fmt.successful_payloads
-            if ret["payload_code"] != SUCCESS_CODE:
-                return False, fmt.payload_error(ret, f"{prefix+5}: " + messages[prefix + 5]), fmt.successful_payloads
-            fmt.add_successful('dismount_storage_file', ret)
+        ret = rcc.run(payloads['attach_storage_file'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+8}: " + messages[prefix + 8]), fmt.successful_payloads
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f"{prefix+9}: " + messages[prefix + 9]), fmt.successful_payloads
+        fmt.add_successful('attach_storage_file', ret)
 
-        return True, "", fmt.successful_payloads
+        return True, messages[1001], fmt.successful_payloads
 
     status, msg, successful_payloads = run_host(host, 3020, {})
-    if status is False:
-        return status, msg
 
-    return True, messages[1000]
+    return status, msg, successful_payloads
```

Comment:
In all other primitives, we are returning a tuple with status boolean and message. We should be consistent, so should we always return the successful_payloads?

Reply:
Hmm. I did this on a bit of a whim, figuring it might make sense to not waste the information in `successful_payloads`. On second thought, that'll cause a lot of confusion further upstream. So yeah, I'll get rid of it. I'm in the middle of the snapshot_hyperv primitive right now, but 'll give this primitive (and test script) a proper once-over to remove the extraneous return value tomorrow.

Reply:
Ok, done now.